#! /bin/bash

#########################################################
# 
# Platform: UQ Flashlite 
# Description: 
#	Trinity de novo transcriptome assembly.
#	This PBS script is part of the Staged Trinity workflow
#	that executes Trinity de novo transcriptome
# 	assembly on University of Queensland's Flashlite HPC.
# Usage: 
#	Edit the script replacing everything in <> with your
#	inputs and options. For very large assemblies (e.g. global
# 	mammalian transcriptomes), increase memory by setting
# 	#PBS -l select=1:ncpus=24:mem=750gb and mem='750G'
# Singularity containers:
#	https://data.broadinstitute.org/Trinity/TRINITY_SINGULARITY/
#
# If you use this script towards a publication, please acknowledge the
# Sydney Informatics Hub (or co-authorship, where appropriate).
#
# Suggested acknowledgement:
# The authors acknowledge the scientific and technical assistance 
# <or e.g. bioinformatics assistance of > of Sydney Informatics Hub, the 
# University of Sydney and resources and services provided by the 
# University of Queensland.
# 
#########################################################

#PBS -A <account>
#PBS -l walltime=24:00:00
#PBS -l select=1:ncpus=24:mem=500gb
#PBS -N Trinity_job1

#Stop script if it encounters an error.
set -e

module add singularity/3.5.1

#Set mem and 
mem='120G'
NCPUS=272

## SET VARIABLES: REPLACE 

#Set home folders needed by dug and singularity
myhome=/d/home/sihtri/sihtri_nathanielb/
dughome=/d/sw/Modules/3.2.7/init/

#Tissue sample name of fastq files
sample=8625

#The top level dir of where scripts, input data, and output data should go.
inpath=/p9/mcc_sihtri/nat

#Singularity Image name
simg=${inpath}/trinity.img

# Grab trinity version for austerity
ver=`echo ${simg} | perl -pe 's/^.*trinityrnaseq\.(v.*)\.simg/$1/g'`
#Set output path
sampleout=Woylie_${sample}_trinity_${ver}
outpath=${inpath}/${sampleout}

#Name of data inputs
first=${inpath}/Woylie/Woylie_8625_S1_R1_trimmed_P.fastq.gz
#,${inpath}/Woylie/woylie_heart_S6_R1_trimmed_P.fastq.gz,${inpath}/Woylie/woylie_pskin_S1_R1_trimmed_P.fastq.gz,${inpath}/Woylie/woylie_tongue_S2_R1_trimmed_P.fastq.gz
second=${inpath}/Woylie/Woylie_8625_S1_R2_trimmed_P.fastq.gz
#,${inpath}/Woylie/woylie_heart_S6_R2_trimmed_P.fastq.gz,${inpath}/Woylie/woylie_pskin_S1_R2_trimmed_P.fastq.gz,${inpath}/Woylie/woylie_tongue_S2_R2_trimmed_P.fastq.gz



# Run trinity, stop after inchworm, do not run chrysalis
singularity exec -B ${dughome} -B ${myhome} -B ${in} ${simg} Trinity \
		--seqType fq \
                --max_memory ${mem} \
                --left ${first} \
                --right ${second} \
                --no_normalize_reads \
		--CPU ${NCPUS} \
                --output ${LOCALOUT} \
                --verbose \
		--no_run_chrysalis

echo "$(date): Finished trinity 1: Jellyfish and Inchworm"
